system/jlib/jlog.cpp

```diff
@@ -1078,6 +1079,15 @@ char const * BinLogMsgHandler::disable()
     return filename.get();
 }
 
+HANDLE BinLogMsgHandler::queryHandle() const
+{
+    CFileIO *cfile = QUERYINTERFACE(fio.get(), CFileIO);
```

Comment:
I can see why you're doing this, but it isn't very nice!

Reply:
agree, the alternative is to expose a queryHandle method in IFileIO which those that support return.